frontend/src/api/RestaurantApi.tsx

```diff
@@ -15,6 +15,7 @@ export const useSearchRestaurants = (
     const accessToken = await getAccessTokenSilently();
     const params = new URLSearchParams();
     params.set('searchQuery', searchState.searchQuery);
+    params.set('page', searchState.page.toString());
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/40#issuecomment-2406936154>#83637f</a><span> - 10/11/2024, 08:53 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The page parameter is set without any validation, which could lead to potential security vulnerabilities or unexpected behavior if an invalid value is provided.
</div>
<div id="fix">
<b>Fix:</b> Implement input validation for the page parameter before setting it in the URL params. Ensure it's a positive integer and within a reasonable range.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -15,7 +15,10 @@
      const accessToken = await getAccessTokenSilently();
      const params = new URLSearchParams();
      params.set('searchQuery', searchState.searchQuery);
 -    params.set('page', searchState.page.toString());
 +    const page = Math.max(1, Math.floor(Number(searchState.page)));
 +    if (isFinite(page)) {
 +      params.set('page', page.toString());
 +    }
 
      const response = await fetch(
        `${API_BASE_URL}/api/restaurant/search/${city}?${params.toString()}`,
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged